simulator/gdb/sim_wrapper.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * sim_wrapper.cpp - Functional simulator wrappers for GDB
+ * @author Vyacheslav Kompan
+ * Copyright 2018 MIPT-MIPS
+ */
+
+#include "sim_wrapper.h"
+
+/* MIPT-MIPS simulator interfaces */
+#include <simulator.h>
+#include <infra/config/config.h>
+#include <mips/mips.h>
+#include <func_sim/func_sim.h>
+
+#define ISA_NAME "mips32"
+
+typedef struct {
+    std::unique_ptr<Simulator> ptr = nullptr;
+    std::string filename;
+} SimulatorInstance;
+static SimulatorInstance simInstance;
+
+
+/* Create simulator */
+void mips_sim_create (const char **argv, const char *filename) try {
+    /* Get argc
+     * Passed arguments start at argv[2], end with NULL */
+    int argc = 0;
+    while (argv[2 + argc++]);
```

Comment:
What's that?

Reply:
GDB doesn't provide `argc` required for arguments parsing, so we have to count it

Reply:
Then it should be a separate function to get `argc` out of `argv`. This piece of code is extremely unobvious, please keep loop condition pure and put all modifications into the body loop.